lib/Sema/TypeChecker.cpp

```diff
@@ -404,6 +405,36 @@ static void bindExtensionDecl(ExtensionDecl *ED, TypeChecker &TC) {
 void TypeChecker::bindExtension(ExtensionDecl *ext) {
   ::bindExtensionDecl(ext, *this);
 }
+void TypeChecker::resolveExtensionForConformanceConstruction(
```

Comment:
I've made this more complicated, using `resolveType` and `ResolveStructure` rather than `validateType`. Thoughts?